File: 22.Book/Book.java
/*Q: Define a class named "Book" having title, author, publisher, qty and price as
data members. Add setter and getter methods inside the class. Define a method named valuation() to calculate the total valuation of all the books. 
 
Provide a test class to test the functionalit of Book class.*/
import java.util.Scanner;
class BookMain
{
	String title,author,publisher;
	int quantity,price;
	void setBook()
	{
		BookMain b1=new BookMain();
		Scanner sc=new Scanner(System.in);
		System.out.print("Enter Title of Book : ");
		title=sc.nextLine();
		System.out.print("Enter Author of Book : ");
		author=sc.nextLine();
		System.out.print("Enter Publisher of Book : ");
		publisher=sc.nextLine();
		System.out.print("Enter Quantity of Book : ");
		quantity=sc.nextInt();
		System.out.print("Enter Price of Book : ");
		price=sc.nextInt();
		
	}
	void getBook()
	{
		System.out.println("Title of Book : "+title);
		
		System.out.println("Author of Book : "+author);
		
		System.out.println("Publisher of Book : "+publisher);
		
		System.out.println("Quantity of Book : "+quantity);
		
		System.out.println("Price of Book : "+price);
		
	}
	double valuation()
	{
		return quantity*price;
	}
}
class Book
{
	public static void main(String args[])
	{
		BookMain b1=new BookMain();
		b1.setBook();
		b1.getBook();
		System.out.println("Total Price of Books : "+b1.valuation());
	}
}

File: ((((Practice Problems))))/Weak Student/CPP/8.Book/Book.cpp
#include<iostream>
using namespace std;

class Book
{
public:
    int isbnn,qty,price;
    char title[20],author[20],publisher[20];

    void setBook()
    {
        cout<<"\nEnter ISBN : ";
        cin>>isbnn;
        cout<<"Enter Title : ";
        cin>>title;
        cout<<"Enter Author : ";
        cin>>author;
        cout<<"Enter Publisher : ";
        cin>>publisher;
        cout<<"Enter Quantity : ";
        cin>>qty;
        cout<<"Enter Price : ";
        cin>>price;
    }

    void getBook()
    {
        cout<<"\nISBN : "<<isbnn<<endl;
        cout<<"Title : "<<title<<endl;
        cout<<"Author : "<<author<<endl;
        cout<<"Publisher : "<<publisher<<endl;
        cout<<"Quantity : "<<qty<<endl;
        cout<<"Price : "<<price<<endl;
    }

    void updateQty()
    {
        cout<<"\nEnter New Quantity : ";
        cin>>qty;
    }

    void updatePrice()
    {
        cout<<"\nEnter New Price : ";
        cin>>price;
    }

    void calBook()
    {
        cout<<"\nTotal Amount of Books is : "<<(qty*price)<<endl;
    }
};

int main()
{
    int choice;
    Book b;
    cout<<"(1).Set Book"<<endl;
    cout<<"(2).Display Book"<<endl;
    cout<<"(3).Update Quantity"<<endl;
    cout<<"(4).Update Price"<<endl;
    cout<<"(5).Total Amount"<<endl;
    cout<<"(6).Exit"<<endl;
    do
    {
        cout<<"\nEnter Choice : ";
        cin>>choice;
    switch(choice)
    {
    case 1:
        b.setBook();
        break;

    case 2:
        b.getBook();
        break;
    case 3:
        b.updateQty();
        break;
    case 4:
        b.updatePrice();
        break;
    case 5:
        b.calBook();
        break;
    case 6:
        break;
    default:
        cout<<"Wrong Input";
        break;
    }
    }while(choice!=6);
}
